pyproject.toml

```diff
@@ -265,3 +265,59 @@ force-exclude = '''
         url = ".tmp/py39-test-image-mpldev/results/fig_comparison.html"
         message = "Click details to see the figure test comparisons for py39-test-image-mpldev."
         report_on_fail = true
+
+[tool.codespell]
+skip = "*.dat,*.fits,*.hdr,*.xml,*egg*,*extern,.git,.tox,_build,fitshdr.htest_groups.py,venv,_*.c,wcs.h,lin.h,tab.h,spc.h,cython*"
+ignore-words-list = """
```

Comment:
I believe that most of these came from variable names, which I didn't want to change without having an understanding of what the variable actually meant.  A few (like `milimeter`) seemed like intentional misspellings. In general, if I wasn't sure about it, I didn't attempt to change it.  

When I applied codespell to PlasmaPy for the first time, it ended up picking up a bunch of false positives from a graphic embedded in a pre-executed Jupyter notebook, so there can be weird reasons like that too!

Reply:
OK, I think it is fine to ignore this for now, but maybe add a comment that some perhaps should be checked and removed? Does `codespell` have some way to tell it to ignore a line? (something like `# noqa: codespell` or so!?)

Reply:
Unfortunately it does not currently have that functionality. I think there's a way to ignore a line by it's number, but that's obviously very unstable and not a good solution.

Reply:
Yes, some of those unit names are from `units` tests for suggestions on misspelled units.

Reply:
>  but maybe add a comment that some perhaps should be checked and removed? 

I think that's a good solution for now.

Reply:
Maybe it is overkill to comb the code.... It cannot detect code snippet automatically (like doctest) and skip those?

Reply:
I agree that I wonder whether it really makes sense to turn this on. On the other hand, the changes here were almost all fine, so perhaps it is OK?

Reply:
We can set up a regex to detect ``>>>`` code snippets. It's not well documented, as demonstrated by https://github.com/codespell-project/codespell/issues/391. I'd suggest this for a followup. The ignore list is not ideal long-term, but feels fine for now.

Reply:
Indeed, let's not have perfect be the enemy of good -- quite a few genuine spelling mistakes caught here!

Reply:
> We can set up a regex to detect >>> code snippets. It's not well documented, as demonstrated by https://github.com/codespell-project/codespell/issues/391. I'd suggest this for a followup. Having an ignore list is not ideal long-term as it allows for the misspells to happen anywhere in the code, but it's still a net improvement over the current situation where all misspells can happen everywhere.

Codespell is designed specifically with source code in mind, so I'd recommend against skipping code snippets.  Codespell generates extremely few false positives because ["it does not check for word membership in a complete dictionary, but instead looks for a set of common misspellings."](https://github.com/codespell-project/codespell#codespell) This makes it highly suitable for CI, with the tradeoff that it will not catch every misspelling.

Reply:
> OK, I think it is fine to ignore this for now, but maybe add a comment that some perhaps should be checked and removed?

Instead of adding a comment, I'd prefer to raise an issue on GitHub after this PR and #13985 are merged to revisit these misspellings and shorten what's in `ignore-words-list`.

Reply:
Sure, that's fine too (though my ideal would be to do both!).

Reply:
Does pyproject.toml even allow comments?

Reply:
Ok. Yes it does. https://toml.io/en/

Reply:
Should we consider moving this list to a new file?

Reply:
> Should we consider moving this list to a new file?

I was wondering the same thing, since another option would be to put this in `.codespellrc` instead.  My inclination would be to keep concentrating configuration information in `pyproject.toml` so that it's all in one place and so that there's less clutter in the top-level directory of the repo, while keeping the codespell config near the bottom of the file.

Reply:
The main reason I suggested a specific word file, was that needing to add/remove a word would "look" like an edit to the installation files.

I do not like having yet another configuration file in the top level (why can't all these things agree to go in a `.config` directory or something?). So I agree keeping it in the `pyproject.toml` file is best.

Reply:
> why can't all these things agree to go in a .config directory or something?

That would be nice.

> So I agree keeping it in the pyproject.toml file is best.

Same.
Resolving this issue. If the word list ever becomes cumbersome or frequently edited we can revisit this issue.